src/commands/scaffold.ts

```diff
@@ -0,0 +1,312 @@
+    const userProjects = await props.apiClient.listProjects({
+      limit: 10,
+    });
+    const selectedProject = await prompts({
+      type: 'select',
+      name: 'value',
+      message: 'Select your Neon project',
+      choices: [
+        { title: 'New', description: 'Create new project', value: false },
+        ...userProjects.data.projects.map((project: any) => ({
+          title: project.name,
+          description: project.id,
+          value: project.id,
+        })),
+      ],
+      initial: false,
+    });
+
+    let projectId = selectedProject.value;
+    if (selectedProject.value === false) {
+      projectData = await createProject(props);
+      projectId = projectData.project.id;
+    } else {
+      projectData = (await props.apiClient.getProject(projectId)).data;
+      const branches: any = (
+        await props.apiClient.listProjectBranches(projectId)
+      ).data.branches;
+      const roles: any = (
+        await props.apiClient.listProjectBranchRoles(projectId, branches[0].id)
+      ).data.roles;
+
+      const connectionString: any = (
+        await props.apiClient.getConnectionUri({
+          projectId: projectId,
+          database_name: branches[0].name,
+          role_name: roles[0].name,
+        })
+      ).data.uri;
+      const connectionUrl = new URL(connectionString);
+
+      projectData.connection_uris = [
+        {
+          connection_uri: connectionString,
+          connection_parameters: {
+            database: connectionUrl.pathname.slice(1),
+            role: connectionUrl.username,
+            password: connectionUrl.password,
+            host: connectionUrl.host,
+          },
+        },
+      ];
+    }
+  }
+
+  if (!props.templateId) {
+    const selectedTemplate = await prompts({
+      type: 'select',
+      name: 'value',
+      message: 'Select your Neon template',
+      choices: availableTemplates.map((template: any) => ({
+        title: template,
+        value: template,
+      })),
+    });
+
+    props.templateId = selectedTemplate.value;
+  }
+
+  if (!availableTemplates.includes(props.templateId)) {
+    log.error(
+      'Template not found. Please make sure the template exists and is public.',
+    );
+    return;
+  }
+
+  let config = null;
+  try {
+    config = await (
+      await getFileContent(
+        REPOSITORY_OWNER,
+        REPOSITORY,
+        (props.templateId as string) + '/config.neon.json',
+      )
+    ).json();
+  } catch (e) {
+    log.error(
+      "Couldn't fetch template config file. Please make sure the template exists and is public.",
+    );
+    log.error(e);
+    return;
+  }
+
+  const dir = path.join(
+    process.cwd(),
+    props.outputDir
+      ? props.outputDir
+      : GENERATED_FOLDER_PREFIX + (projectData.project.name as string),
+  );
+  if (props.templateId) {
+    await downloadFolderFromTree(
+      REPOSITORY_OWNER,
+      REPOSITORY,
+      'main',
+      props.templateId,
+      dir,
+    );
+  } else {
```

Comment:
If this a possible code path? seems like we are returning early in line 250 if the templateID doesn't exist?

Reply:
Yes, you are right, this is dead code branch. Removed it.